tests/integration/src/test/scala/mesosphere/marathon/integration/AppDeployIntegrationTest.scala

```diff
@@ -671,6 +671,13 @@ class AppDeployIntegrationTest extends AkkaIntegrationTest with EmbeddedMarathon
       Then("the deployment gets created")
       WaitTestSupport.validFor("deployment visible", 1.second)(marathon.listDeploymentsForPathId(id).value.size == 1)
 
+      Then("overdue flag is set to true")
+      eventually {
+        val queueItem = marathon.launchQueue().value.queue.find(_.app.id equals id.toString)
```

Comment:
also verified that this new test behavior fails with the current HEAD.

Reply:
You mean this PR doesn't fix the borked behavior?

Reply:
no i meant to say that if I revert the source code changes but not the test changes from this PR - then the test would fail.

Reply:
Ah ok cool – I just wanted to make sure I understand correctly :)